include/pqxx/internal/conversions.hxx

```diff
@@ -1156,6 +1156,37 @@ inline constexpr format param_format(std::vector<std::byte, Args...> const &)
 template<typename T> inline constexpr bool is_sql_array<std::vector<T>>{true};
 
 
+#if defined(PQXX_HAVE_SPAN) && __has_include(<span>)
+template<typename T, size_t Extent>
+struct nullness<std::span<T, Extent>> : no_null<std::span<T, Extent>>
+{};
+
+
+template<typename T, size_t Extent>
+struct string_traits<std::span<T, Extent>>
+        : internal::array_string_traits<std::span<T, Extent>>
+{};
+
+
+template<typename T, size_t Extent>
+inline constexpr format param_format(std::span<T, Extent> const &)
+{
+  return format::text;
+}
+
+
+template<size_t Extent>
+inline constexpr format param_format(std::span<std::byte, Extent> const &)
+{
+  return format::binary;
```

Comment:
I wonder if we could get away with returning whatever `param_format()` would return for the contained type.

Reply:
It is `text` by default, is it not?

Reply:
That's right.